estate/models/estate_property.py

```diff
@@ -0,0 +1,103 @@
+from odoo import api, fields, models
+from odoo.exceptions import UserError, ValidationError
+from odoo.tools import float_compare
+
+DEFAULT_GARDEN_AREA = 10
+DEFAULT_GARDEN_ORIENTATION = "north"
+
+
+class PropertyModel(models.Model):
+    _name = "estate.property"
+    _description = "Estate Property model"
+    _order = "id desc"
+
+    name = fields.Char("Title", required=True)
+    description = fields.Text()
+    postcode = fields.Char()
+    date_availability = fields.Date(default=fields.Date.add(fields.Date.today(), months=3), copy=False)
+    expected_price = fields.Float(required=True)
+    best_offer = fields.Float(compute="_compute_highest_price")
+    selling_price = fields.Float(readonly=True, copy=False)
+    bedrooms = fields.Integer(default=2)
+    living_area = fields.Integer("Living Area (sqm)")
+    facades = fields.Integer()
+    garage = fields.Boolean()
+    garden = fields.Boolean()
+    garden_area = fields.Integer("Garden Area (sqm)")
+    garden_orientation = fields.Selection(
+        selection=[('north', 'North'), ('south', 'South'), ('east', 'East'), ('west', 'West')]
+    )
+    total_living_area = fields.Integer("Total Area (sqm)", compute="_compute_total_area")
+
+    active = fields.Boolean(default=True)
+    state = fields.Selection(
+        selection=[
+            ("new", "New"),
+            ("received", "Offer Received"),
+            ("accepted", "Offer Accepted"),
+            ("sold", "Sold"),
+            ("cancelled", "Cancelled")
+        ],
+        string="Status",
+        required=True,
+        copy=False,
+        default="new"
+    )
+    property_type_id = fields.Many2one("estate.property.type")
+    buyer = fields.Many2one("res.partner", copy=False)
+    salesperson = fields.Many2one("res.users", default=lambda self: self.env.user)
```

Comment:
Should contain _id in the name as they are relationship fields.
```suggestion
    buyer_id = fields.Many2one("res.partner", copy=False)
    salesperson_id = fields.Many2one("res.users", default=lambda self: self.env.user)
```